### PR TITLE
NEW Seperate tx pull and push commands

### PR DESCRIPTION
**Pull request by emteknetnz**

### CODE REVIEW

File: src/Steps/Release/UpdateTranslations.php [hunk 1]
```diff
@@ -295,16 +310,12 @@ protected function generateJavascript(OutputInterface $output, $modules)
      * @param OutputInterface $output
      * @param Module[] $modules
      */
-    public function pushSource(OutputInterface $output, $modules)
+    public function transifexPushSource(OutputInterface $output, $modules)
```
Comment (maxime-rainville) [inline src/Steps/Release/UpdateTranslations.php:313]: There's no outside reference to the method. Might as well make it protected to make clear we don't expect outside classes to call it.
```suggestion
    protected function transifexPushSource(OutputInterface $output, $modules)
```

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

File: src/Commands/Release/Release.php [hunk 2]
```diff
@@ -45,6 +45,18 @@ protected function configureOptions()
...
+                null,
+                InputOption::VALUE_NONE,
+                'Skip running tx pull, i18nTextCollectorTask and js/json updates'
+            )
+            ->addOption(
+                'i18n-tx-push',
+                null,
+                InputOption::VALUE_NONE,
+                'Run tx push -s - only do this during the beta1 release of a new minor'
+            )
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply src/Commands/Release/Release.php:59]: >> What's the expected outcome if you provide both those flags simultaneously?

documented in description:
```
push only (unlikely to ever use this)
cow release:translate -vvv --i18n-tx-push --skip-i18n-tx-pull-and-update --skip-fetch-tags 4.10.0-beta1 silverstripe/installer
```

>> I wouldn't be able to tell what they do without looking a the source code / I would rather the flags describe what's happening rather than the actual operations that are being run

Also I think that `i18n-tx-pull-and-update` is currently more descriptive than --translation-operation:pull, because it also `updates` meaning it runs the text collector and does a bunch of stuff to update the files

Does naming really even matter?  The intention is really that we just copy paste instructions the correct instructions out of confluence doc.  Any idea what this flag does without looking at the source code? `skip-emulate-requirements`.  I have no idea.

Basically the idea of this approach is this:
- without flag, do a pull because it's safe
- if you want to push, put a flag in because it might be unsafe

File: src/Commands/Release/Release.php [hunk 3]
```diff
@@ -118,6 +130,10 @@ protected function fire()
         // Update all translations
         if (!$this->input->getOption('skip-i18n')) {
             $translate = new UpdateTranslations($this, $project, $releasePlan);
+            $doTxPullAndUpdate = !$this->input->getOption('skip-i18n-tx-pull-and-update');
+            $translate->setDoTransifexPullAndUpdate($doTxPullAndUpdate);
+            $doTxPush = $this->input->getOption('i18n-tx-push');
+            $translate->setDoTransifexPush($doTxPush);
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply src/Commands/Release/Release.php:136]: Seems better to have all the "CLI flag" related stuff in one place, rather than splitting it across multiple files.  Unless we do that already?

File: src/Steps/Release/UpdateTranslations.php (hunk 1 above)
Comment (emteknetnz) [reply src/Steps/Release/UpdateTranslations.php:313]: Updated

File: src/Commands/Release/Release.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (maxime-rainville) [reply src/Commands/Release/Release.php:59]: I knew up-front what you were trying to implement and I had to read the thing a few times to figure out what combination of parameters I was meant to use to achieve a given results.

Yes it's an internal tool, but we should aim to make it easy for us to understand what it's doing both to minimise the risk of error and to make it easier to onboard new devs.

Seems to me like it's worth spending 5 min to make this clear and concise and save us pointless doc reading on every follow up release.

File: src/Commands/Release/Release.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxime-rainville) [reply src/Commands/Release/Release.php:136]: Just reviewed how we use `$input->getOption()` in the rest of Cow and it seems pretty inconsistent as-is.

This approach is fine

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

File: src/Commands/Release/Release.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (emteknetnz) [reply src/Commands/Release/Release.php]: I changed `tx` to `translation`
